packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-timeline/atomic-insight-user-actions-timeline.tsx

```diff
@@ -0,0 +1,222 @@
+import {
+  buildUserActions,
+  UserActions,
+  UserActionsState,
+  UserSession,
+} from '../..';
+import ArrowDownIcon from '../../../../images/arrow-full-down.svg';
+import ArrowUpIcon from '../../../../images/arrow-up.svg';
+import {
+  InitializableComponent,
+  InitializeBindings,
+  BindStateToController,
+} from '../../../../utils/initialization-utils';
+import {NoItemsContainer} from '../../../common/no-items/container';
+import {MagnifyingGlass} from '../../../common/no-items/magnifying-glass';
+import {InsightBindings} from '../../atomic-insight-interface/atomic-insight-interface';
+
+/**
+ * @internal
+ * This component displays all the actions performed by a user around the time they created a case.
+ * The actions are grouped into multiple sessions, including the session during which the case was created,
+ * the sessions preceding the case creation, and the sessions following the case creation.
+ *
+ * @component
+ * @example
+ * <AtomicInsightUserActionsTimeline userId={'123'} caseCreationDate={'2024-08-15T10:00:00Z'} />
+ *
+ */
+@Component({
+  tag: 'atomic-insight-user-actions-timeline',
+  styleUrl: 'atomic-insight-user-actions-timeline.pcss',
+  shadow: true,
+})
+export class AtomicInsightUserActionsTimeline
+  implements InitializableComponent<InsightBindings>
+{
+  @InitializeBindings() public bindings!: InsightBindings;
+  public userActions!: UserActions;
+
+  @BindStateToController('userActions')
+  @State()
+  public userActionsState!: UserActionsState;
+  public error!: Error;
+
+  /**
+   * The ID of the user whose actions are being displayed.
+   */
+  @Prop() public userId!: string;
+  /**
+   * The date and time when the case was created..
+   */
+  @Prop() public ticketCreationDate!: string;
+
+  public initialize() {
+    this.userActions = buildUserActions(this.bindings.engine, {
+      options: {ticketCreationDate: this.ticketCreationDate},
+    });
+
+    this.userActions.fetchUserActions(this.userId);
+  }
+
+  @State() followingSessionsAreVisible = false;
+  @State() precedingSessionsAreVisible = false;
+
+  private toggleFollowingSessions() {
+    this.followingSessionsAreVisible = !this.followingSessionsAreVisible;
+  }
+
+  private togglePrecedingSessions() {
+    this.precedingSessionsAreVisible = !this.precedingSessionsAreVisible;
+  }
+
+  private renderSessions(
+    sessions: Array<UserSession> | undefined,
+    renderSeparator?: Function,
+    testId?: string
+  ) {
+    return (
+      <div>
+        {sessions?.map(({actions, start}) => [
+          <div class="mt-4">
+            <atomic-insight-user-actions-session
+              startTimestamp={start}
+              userActions={actions}
+              data-testid={testId}
+            ></atomic-insight-user-actions-session>
+          </div>,
+          renderSeparator ? renderSeparator() : null,
+        ])}
+      </div>
+    );
+  }
+
+  private renderToggleFollowingSessionsButton() {
```

Comment:
same here, I would put renderTogglePrecedingSessionsButton before following in the file